qcodes/station.py

```diff
@@ -217,3 +251,212 @@ def __getitem__(self, key):
+        # and setting existing one
+        def setup_parameter_from_dict(parameter: Parameter, options_dict):
+            for attr, val in options_dict.items():
+                if attr in PARAMETER_ATTRIBUTES:
+                    # set the attributes of the parameter, that map 1 to 1
+                    setattr(parameter, attr, val)
+                # extra attributes that need parsing
+                elif attr == 'limits':
+                    lower, upper = [float(x) for x in val.split(',')]
+                    parameter.vals = validators.Numbers(lower, upper)
+                elif attr == 'monitor' and val is True:
+                    self._monitor_parameters.append(parameter)
+                elif attr == 'alias':
+                    setattr(instr, val, parameter)
+                elif attr == 'initial_value':
+                    # skip value attribute so that it gets set last
+                    # when everything else has been set up
+                    pass
+                else:
+                    log.warning(f'Attribute {attr} not recognized when'
+                                f' instatiating parameter \"{parameter.name}\"')
+            if 'initial_value' in options_dict:
+                parameter.set(options_dict['initial_value'])
+
+        def resolve_parameter_identifier(instrument: Instrument,
+                                         identifier: str) -> Parameter:
+            p = instrument
+            for level in identifier.split('.'):
+                p = getattr(p, level)
+            if not isinstance(p, Parameter):
+                raise RuntimeError(
+                    f'Cannot resolve parameter identifier `{identifier}` to '
+                   f'a parameter on instrument {instrument!r}.')
+            return cast(Parameter, p)
+
+        # setup existing parameters
+        for name, options in instr_cfg.get('parameters', {}).items():
+            setup_parameter_from_dict(
+                resolve_parameter_identifier(instr, name),
+                options)
+
+        # setup new parameters
+        for name, options in instr_cfg.get('add_parameters', {}).items():
+            # allow only top level paremeters for now
+            # pop source only temporarily
+            source = options.pop('source', False)
+            if source:
+                source_p = resolve_parameter_identifier(instr, source)
+                instr.add_parameter(name,
+                                    DelegateParameter,
+                                    source=source_p)
+            else:
+                instr.add_parameter(name, Parameter)
+            p = getattr(instr, name)
+            setup_parameter_from_dict(p, options)
+            # restore source
+            options['source'] = source
+
+        # add the instrument to the station
+        self.add_component(instr)
+
+        # restart Monitor
+        Monitor(*self._monitor_parameters)
```

Comment:
Can this be made optional? I don't think it's fair to fire up a `Monitor` in case the user didn't want it.

Also, we seem to need to incorporate this with the `monitor` argument of the `Station` `__init__`. I do not know what is the intent and history of it, so can't suggest anything. Also `Monitor` is not used in Delft, so I can't suggest what the best user experience should be. What I know for sure is that we need to make this optional. Maybe the following would work in the short term: if there are no parameters to monitor from yaml file, the `Monitor` is not initiated.

Another reason to make this step optional somehow is: what if a user already had a Monitor running (for some reason), and now he loads stuff from yaml station configuration, and all the parameters which he had set before are gone.

Reply:
Fair points.... I will think of something to make the monitor part optional.

Reply:
ok how about that: we make the monitor init kwarg an optional boolean. When true a monitor is created by the station and gets repopulated with the station `_monitor_parameters`. When false, nothing happens, and when None the value is read from qcodesrc.json.
The situation with the monitor is far from ideal but I don't want to go down that rabbit hole. To start with the parameters are not public and on runtime no parameters can be added...

Reply:
so you are proposing the following kwarg, right?
* `monitor == True` - `Station` creates a monitor when loading instruments from config in case there are parameters to monitor.
* `monitor == False` - `Station` does nothing monitor-related
* `monitor == None` - the True/False is read from qcodesrc.json

I agree with this as a solution for now.

Clarification questions and suggestions:
* what is the default value for `monitor` kwarg? I suggest `None`.
* what is the default value of `monitor` in qcodesrc.json?  I suggest to just not have it in the default qcodesrc.json (or if that's not allowed by the schema, let's make it `False` then)
* do we need to make a settable `monitor` property on the Station for dynamic adjustments of the station's monitor-related behavior? i propose to not do that for now, let's see if we get requests in this direction from users.